### PR TITLE
Implement Oauth Popup UI C-444

### PR DESCRIPTION
**Pull request by nicoback2**
### Description
Screen Shot 2022-05-19 at 12 48 09 PM

Screen Shot 2022-05-19 at 12 48 03 PM

Screen Shot 2022-05-19 at 12 12 59 PM

Designs: https://www.figma.com/file/D2mxhTtj9MB0ex2f6Osv9t/%F0%9F%94%97-OAuth-V1-(Read-Only)?node-id=0%3A1

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

Tested in browser (/oauth/auth?scope=read&app_name=Demo&redirect_uri=https://facebook.com)

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### CODE REVIEW

**Review by sliptype — APPROVED** [review]
Looks great to me! A tiny nit maybe to increase the line height for the redirect location notice

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--oauth-popup-ui

File: packages/web/src/pages/oauth-login-page/OAuthLoginPage.tsx [hunk 1]
```diff
@@ -27,12 +99,39 @@ export const OAuthLoginPage = () => {
...
-  const [submitError, setSubmitError] = useState<string | null>(null)
+  const [userEmail, setUserEmail] = useState<string | null>(null)
+  useEffect(() => {
+    const getAndSetEmail = async () => {
+      let email: string
+      try {
+        email = await AudiusBackend.getUserEmail()
+      } catch {
+        setUserEmail(null)
+        window.location.href = ERROR_PAGE
```
Comment (dylanjeffers) [inline packages/web/src/pages/oauth-login-page/OAuthLoginPage.tsx:112]: not sure what the standard approach is here, but thinking we should ideally use `react-router` to push to error_page?

File: packages/web/src/pages/oauth-login-page/OAuthLoginPage.tsx [hunk 2]
```diff
@@ -64,18 +163,33 @@ export const OAuthLoginPage = () => {
...
+    queryParamsError = messages.scopeError
+  }
+
+  const formOAuthResponse = async (account: User) => {
+    let email: string
+    if (!userEmail) {
+      try {
+        email = await AudiusBackend.getUserEmail()
+      } catch {
+        return
```
Comment (dylanjeffers) [inline packages/web/src/pages/oauth-login-page/OAuthLoginPage.tsx:187]: would we want to also go to error page here like in the useEffect?

File: packages/web/src/pages/oauth-login-page/OAuthLoginPage.tsx [hunk 3]
```diff
@@ -98,111 +212,259 @@ export const OAuthLoginPage = () => {
...
     e.preventDefault()
+    clearErrors()
+    setIsSubmitting(true)
     let signInResponse: any
     try {
-      signInResponse = await AudiusBackend.signIn(email, password)
+      signInResponse = await AudiusBackend.signIn(emailInput, passwordInput)
     } catch (err) {
-      setSubmitError('Unknown error')
+      setIsSubmitting(false)
```
Comment (dylanjeffers) [inline packages/web/src/pages/oauth-login-page/OAuthLoginPage.tsx:243]: are there any cases where we would potentially override `isSubmitting` and `generalSubmitError` with two in-flight async calls? reason i ask, is that we could maybe use `useAsync` and `useAsyncFn` to define these callbacks/effects which would isolate the loading and error logic. down to chat about those options.

**Review by dylanjeffers — APPROVED** [review]
Looks great, also small callout, we use formik on native for some form work, wondering if it could help simplify some of the form logic here (prob not too much since this is a simple form, but might be worth it if you think)

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--oauth-popup-ui

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--oauth-popup-ui

**Comment by nicoback2** [comment]
> Looks great, also small callout, we use formik on native for some form work, wondering if it could help simplify some of the form logic here (prob not too much since this is a simple form, but might be worth it if you think)

Ty for review! Yeah I thought about using Formik but decided against it since the form is so small and quite uncomplicated in terms of validation. I've found that Formik over complicates small forms like this

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--oauth-popup-ui

**Pull request merged**